healvis/simulator.py

```diff
@@ -403,15 +527,25 @@ def run_simulation(param_file, Nprocs=1, sjob_id=None, add_to_history=''):
     # SkyModel
     # ---------------------------
     # construct sky model
-    sky = sky_model.construct_skymodel(skyparam['sky_type'], freqs=freq_array, Nside=skyparam['Nside'],
-                                       ref_chan=skyparam['ref_chan'], sigma=skyparam['sigma'], Nskies=Nskies)
-
-    # If loading a healpix map from disk, use those frequencies instead of ones specified in obsparam
-    if skyparam['sky_type'].lower() not in ['flat_spec', 'gsm']:
-        param_dict['freq'] = {'freq_array': sky.freqs}
+    if 'Nskies' not in skyparam:
+        skyparam['Nskies'] = Nskies
+    sky_type = skyparam.pop('sky_type')
+    savepath = None
+    if 'savepath' in skyparam:
+        savepath = skyparam.pop('savepath')
+
+    sky = sky_model.construct_skymodel(sky_type, **skyparam)
+
+    # If loading a healpix map from disk, confirm its frequencies match the obsparam frequencies.
+    if sky_type.lower() not in ['flat_spec', 'gsm']:
+        try:
+            assert np.all(freq_array == sky.freqs)
```

Comment:
I don't like `==` for float data types. I agree that if they were literally the same bytes then this will work, but I think that if they disagreed at 1e-15 level then this should pass, and in the way its written it will fail. I think `np.isclose(arr1, arr2).all()` would be better, personally.